### PR TITLE
Movie service

### PR DESCRIPTION
**Pull request by milutinzeljkovic**

### CODE REVIEW

File: app/Http/Controllers/Auth/RegisterController.php [hunk 1]
```diff
@@ -61,12 +63,12 @@ protected function validator(array $data)
...
      * @return \App\User
      */
-    protected function create()
+    protected function create(RegisterUserRequest $request)
     {
         return User::create([
-            'name' => request('name'),
-            'email' => request('email'),
-            'password' => Hash::make(request('password')),
+            'name' => $request->name,
```
Comment (Pseudonick47) [inline app/Http/Controllers/Auth/RegisterController.php]: request validated

File: app/Http/Requests/MovieStoreRequest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'title' => 'required',
+            'description' => 'required',
+            'image_url' => 'required',
+            'genre_id' => 'required'
```
Comment (miljanic) [inline app/Http/Requests/MovieStoreRequest.php]: exists tabela....

File: app/Movie.php [hunk 3]
```diff
@@ -6,5 +6,5 @@
 
 class Movie extends Model
 {
-    //
+    protected $guarded = ['id', 'genre_id'];
```
Comment (miljanic) [inline app/Movie.php]: moye genre aj di

File: app/Policies/MoviePolicy.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    /**
+     * Determine whether the user can view the movie.
+     *
+     * @param  \App\User  $user
+     * @param  \App\Movie  $movie
+     * @return mixed
+     */
+    public function view(User $user, Movie $movie)
+    {
+        return !$user ? false : true;
```
Comment (miljanic) [inline app/Policies/MoviePolicy.php]: ne treba

File: app/Services/MovieService.php [hunk 5]
```diff
@@ -3,11 +3,25 @@
...
+
+    public function deleteMovie($movie)
+    {
+        return Movie::where('id',$movie->id)->delete();
+    }
+
+    public function updateMovie($movie,$values)
+    {
+        $movie->update($values);
+        return Movie::where('user_id', Auth::id())->where('id',$movie->id)->get();
```
Comment (miljanic) [inline app/Services/MovieService.php]: return movie

File: database/migrations/2019_10_14_113858_alter_movies_table.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('movies', function (Blueprint $table) {
+            $table->integer('user_id');
```
Comment (miljanic) [inline database/migrations/2019_10_14_113858_alter_movies_table.php]: fali foreign key

File: database/migrations/2019_10_14_113858_alter_movies_table.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('movies', function (Blueprint $table) {
+            $table->dropColumn('user_id');
```
Comment (miljanic) [inline database/migrations/2019_10_14_113858_alter_movies_table.php:30]: bice drop foreign

**Review by miljanic — APPROVED** [review]

**Pull request merged**